cmd/internal/ecs/aws_api_names.go

```diff
@@ -0,0 +1,78 @@
+package ecs
+
+import (
+	"context"
+
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/aws/aws-sdk-go-v2/service/ecs"
+)
+
+// Utility method for: listing a bunch of ARNs, and then looking up their names.
+//
+// Example: in a region, list all clusters.
+// Example: in a cluster, list all services.
+//
+
+// Interface of the Paginator returned by the ECS package.  TODO: should we make
+// this take the Options type as an extra parameter, so it can be used with other packages?
+type AWSPaginator[ListOutput any] interface {
+	HasMorePages() bool
+	NextPage(context.Context, ...func(*ecs.Options)) (ListOutput, error)
+}
+
+func ListAWSObjectsByName[
+	ListOutput any, // an SDK output type for the List operation
+	DescribeInput any, // an SDK input type for the Describe operation
+	DescribeOutput any, // an SDK output type for the Describe operation
+	NamedItem any, // The type of the field within the Describe output
+](
+	ctx context.Context,
+	objectName string, // "Task" or "Cluster" etc. for debugging
+	paginator AWSPaginator[ListOutput], // Paginator for the List<Object> call
+	getArns func(ListOutput) []arn, // Convert list output to list of ARNs
+	inputFactory func([]arn) DescribeInput, // Convert list of ARNs to an input to the describe function
+	describe func(context.Context, DescribeInput) (DescribeOutput, error), // SDK function to call Describe<Object>
+	extract func(DescribeOutput) []NamedItem, // Extract a list of ARN/name pairs from the Describe output
+	convert func(NamedItem) (arn, string), // Convert the ARN/name pair to a usable form, return "" if not present
```

Comment:
It was meant to refer to NamedItem, which is a types.Service or types.Cluster.  These are more than just the pair, but that's what we're treating them as.  I didn't want to call it NamedItem again because that's already a horrible name.  :)

Reply:
I changed the comment to "extract a list of objects containing ARN/name pairs from the Describe output".